mephisto/core/task_launcher.py

```diff
@@ -83,13 +96,46 @@ def create_assignments(self) -> None:
                 )
                 self.units.append(Unit(self.db, unit_id))
 
-    def launch_units(self, url: str) -> None:
-        """launch any units registered by this TaskLauncher"""
         for unit in self.units:
+            self.unlaunched_units[unit.db_id] = unit
+
+    def generate_units(self):
+        """ units generator which checks that only 'max_num_concurrent_units' running at the same time,
+        i.e. in the LAUNCHED or ASSIGNED states """
+        while True:
+            for _, launched_unit in self.launched_units.copy().items():
+                if (
+                    launched_unit.get_status() != AssignmentState.LAUNCHED
```

Comment:
`get_status()` incurs at the least a database hit, and at the current most a round trip to determine the status of an MTurk. Prefer getting this value into a variable first and then comparing.